src/parser/ParsedQuery.h

```diff
@@ -138,6 +138,39 @@ class VariableOrderKey {
 // Represents an ordering by a variable or an expression.
 using OrderKey = std::variant<VariableOrderKey, ExpressionOrderKey>;
 
+/// Store an expression that appeared in a GROUP BY clause.
+class ExpressionGroupKey {
+ public:
+  sparqlExpression::SparqlExpressionPimpl expression_;
+  // ___________________________________________________________________________
+  explicit ExpressionGroupKey(
+      sparqlExpression::SparqlExpressionPimpl expression)
+      : expression_{std::move(expression)} {}
+};
+
+/// Store an expression that is bound to a variable and appeared in a GROUP BY
+/// clause.
+class ExpressionAliasGroupKey {
+ public:
+  sparqlExpression::SparqlExpressionPimpl expression_;
+  string variable_;
+  // ___________________________________________________________________________
+  explicit ExpressionAliasGroupKey(
+      sparqlExpression::SparqlExpressionPimpl expression, string variable)
+      : expression_{std::move(expression)}, variable_{std::move(variable)} {}
+};
```

Comment:
Why is this a new type, and not simply an `Alias` (which maybe already exists).

I think we don't need `Alias` and `ExpressionAliasGroup`, these types should be unified.